src/VisualStudio/Core/Def/Packaging/IPackageSearchIOService.cs

```diff
@@ -0,0 +1,23 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.VisualStudio.LanguageServices.Packaging
+{
+    /// <summary>
+    /// Used so we can mock out how the search service does IO for testing purposes.
+    /// </summary>
+    internal interface IPackageSearchIOService
+    {
+        void Create(DirectoryInfo directory);
+        void Delete(FileInfo file);
+        IEnumerable<FileInfo> EnumerateFiles(DirectoryInfo directory);
+        bool Exists(FileSystemInfo info);
```

Comment:
I'd prefer we had methods like `TryOpen`.  If exceptions are too disruptive to the process then we should go with the Try pattern.  But of course that requires BCL changes ...  :frowning: